tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py

```diff
@@ -334,8 +334,12 @@ def main(argv):
     # target proxy cannot deleted unless the forwarding rule is deleted). The
     # leaked target-proxy is guaranteed to be a super set of leaked
     # forwarding-rule.
-    leakedHealthChecks = exec_gcloud(project, 'compute', 'health-checks',
-                                     'list')
+    compute = gcp.compute.ComputeV1(gcp.api.GcpApiManager(), project)
+    r = compute.list_health_check()
+    leakedHealthChecks = [
+        ii for ii in r['items'] if datetime.datetime.fromisoformat(
+            ii['creationTimestamp']) <= get_expire_timestamp()
+    ]
     delete_leaked_td_resources(dry_run, td_resource_rules, project, network,
                                leakedHealthChecks)
```

Comment:
Well, not related to this PR, but the deletion of the-TD-resource-set might be interrupted, then leaked. E.g., health-check is removed, then the cleanup job timeout, leaving url-map, forwarind-rule, target proxy forevery.

I would still recommend to check all costly resources.